grpc compile fail with google-protobuf

**Describe the bug**
I use GRPC and I use to [google protobuf ](https://github.com/protocolbuffers/protobuf/tree/master/src/google/protobuf) like for example [timestamp.proto](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto). I'm my other microservices (spring-boot), I easily include the maven dependency `com.google.protobuf:protobuf-java` and I can use the google proto. But with quarkus it doesn't work out of the box.

**Expected behavior**
I can include the `com.google.protobuf:protobuf-java` dependencies and quakurs grpc extension will pick it up automatically.

**Actual behavior**
Compilation failed with
```
[INFO] --- quarkus-maven-plugin:1.8.3.Final:generate-code (default) @ grpc-google-protobuf ---
google/protobuf/timestamp.proto: File not found.
hello.proto:7:1: Import "google/protobuf/timestamp.proto" was not found or had errors.
hello.proto:12:5: "google.protobuf.Timestamp" is not defined.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.284 s
[INFO] Finished at: 2020-10-14T16:54:56+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.8.3.Final:generate-code (default) on project grpc-google-protobuf: Prepare phase of the quarkus-maven-plugin threw an error: Failed to generate Java classes from proto files: [/home/gaetan/workspace/github/grpc-google-protobuf/src/main/proto/hello.proto] to /home/gaetan/workspace/github/grpc-google-protobuf/target/generated-sources/grpc -> [Help 1]
```

**To Reproduce**

https://github.com/gaetancollaud/grpc-google-protobuf

You can look at the branch [protobuf-maven-plugin](https://github.com/gaetancollaud/grpc-google-protobuf/tree/protobuf-maven-plugin) where I disable the goal  `generate-code` of the `quarkus-maven-plugin` and where I manually configured the `protobuf-maven-plugin`. This version works and I can compile, but then `mvn quarkus:dev` auto reload is broken.

Steps to reproduce the behavior:
1. Checkout the repo https://github.com/gaetancollaud/grpc-google-protobuf
2. `mvn compile`

**Configuration**
*Nothing specific*

**Screenshots**
*No screenshot*

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:  Linux gaetan-desktop 5.4.0-48-generic #52-Ubuntu SMP Thu Sep 10 10:58:49 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: Java version: 11.0.8, vendor: GraalVM Community, runtime: /home/gaetan/.sdkman/candidates/java/20.2.0.r11-grl
 - GraalVM version (if different from Java): 20.2.0.r11
 - Quarkus version or git rev: 1.8.3
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

**Additional context**
*No additional context*

--- a/core/deployment/src/main/java/io/quarkus/deployment/CodeGenContext.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/CodeGenContext.java
@@ -1,6 +1,7 @@
 package io.quarkus.deployment;
 
 import java.nio.file.Path;
+import java.util.Map;
 
 import io.quarkus.bootstrap.model.AppModel;
 
@@ -10,13 +11,16 @@ public class CodeGenContext {
     private final Path workDir;
     private final Path inputDir;
     private final boolean redirectIO;
+    private final Map<String, String> properties;
 
-    public CodeGenContext(AppModel model, Path outDir, Path workDir, Path inputDir, boolean redirectIO) {
+    public CodeGenContext(AppModel model, Path outDir, Path workDir, Path inputDir, boolean redirectIO,
+            Map<String, String> properties) {
         this.model = model;
         this.outDir = outDir;
         this.workDir = workDir;
         this.inputDir = inputDir;
         this.redirectIO = redirectIO;
+        this.properties = properties;
     }
 
     public AppModel appModel() {
@@ -37,5 +41,9 @@ public class CodeGenContext {
 
     public boolean shouldRedirectIO() {
         return redirectIO;
+    }
+
+    public Map<String, String> properties() {
+        return properties;
     }
 }

--- a/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java
@@ -5,6 +5,7 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import java.util.ServiceLoader;
 import java.util.Set;
 import java.util.function.Consumer;
@@ -23,11 +24,11 @@ public class CodeGenerator {
     public static void initAndRun(ClassLoader classLoader,
             Set<Path> sourceParentDirs, Path generatedSourcesDir, Path buildDir,
             Consumer<Path> sourceRegistrar,
-            AppModel appModel) throws CodeGenException {
+            AppModel appModel, Map<String, String> properties) throws CodeGenException {
         List<CodeGenData> generators = init(classLoader, sourceParentDirs, generatedSourcesDir, buildDir, sourceRegistrar);
         for (CodeGenData generator : generators) {
             generator.setRedirectIO(true);
-            trigger(classLoader, generator, appModel);
+            trigger(classLoader, generator, appModel, properties);
         }
     }
 
@@ -75,18 +76,21 @@ public class CodeGenerator {
      * @param deploymentClassLoader deployment classloader
      * @param data code gen
      * @param appModel app model
+     * @param properties custom code generation properties
      * @return true if sources have been created
      * @throws CodeGenException on failure
      */
     public static boolean trigger(ClassLoader deploymentClassLoader,
             CodeGenData data,
-            AppModel appModel) throws CodeGenException {
+            AppModel appModel,
+            Map<String, String> properties) throws CodeGenException {
         return callWithClassloader(deploymentClassLoader, () -> {
             CodeGenProvider provider = data.provider;
 
             return Files.isDirectory(data.sourceDir)
                     && provider.trigger(
-                            new CodeGenContext(appModel, data.outPath, data.buildDir, data.sourceDir, data.redirectIO));
+                            new CodeGenContext(appModel, data.outPath, data.buildDir, data.sourceDir, data.redirectIO,
+                                    properties));
         });
     }
 

--- a/core/deployment/src/main/java/io/quarkus/deployment/codegen/CodeGenData.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/codegen/CodeGenData.java
@@ -19,7 +19,7 @@ public class CodeGenData {
         this.provider = provider;
         this.outPath = outPath;
         this.sourceDir = sourceDir;
-        this.buildDir = buildDir;
+        this.buildDir = buildDir.normalize();
         this.redirectIO = redirectIO;
     }
 

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java
@@ -102,7 +102,7 @@ public class IsolatedDevModeMain implements BiConsumer<CuratedApplication, Map<S
                             }
                         });
 
-                startCodeGenWatcher(deploymentClassLoader, codeGens);
+                startCodeGenWatcher(deploymentClassLoader, codeGens, context.getBuildSystemProperties());
 
                 augmentDone = true;
                 runner = start.runMainClass(context.getArgs());
@@ -143,7 +143,8 @@ public class IsolatedDevModeMain implements BiConsumer<CuratedApplication, Map<S
         }
     }
 
-    private void startCodeGenWatcher(QuarkusClassLoader classLoader, List<CodeGenData> codeGens) {
+    private void startCodeGenWatcher(QuarkusClassLoader classLoader, List<CodeGenData> codeGens,
+            Map<String, String> properties) {
 
         Collection<FSWatchUtil.Watcher> watchers = new ArrayList<>();
         for (CodeGenData codeGen : codeGens) {
@@ -152,7 +153,7 @@ public class IsolatedDevModeMain implements BiConsumer<CuratedApplication, Map<S
                         try {
                             CodeGenerator.trigger(classLoader,
                                     codeGen,
-                                    curatedApplication.getAppModel());
+                                    curatedApplication.getAppModel(), properties);
                         } catch (Exception any) {
                             log.warn("Code generation failed", any);
                         }

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
@@ -89,7 +89,8 @@ public class QuarkusGenerateCode extends QuarkusTask {
                         paths.iterator().next(),
                         buildDir,
                         sourceRegistrar,
-                        appCreationContext.getAppModel());
+                        appCreationContext.getAppModel(),
+                        realProperties);
 
             }
         } catch (BootstrapException | IllegalAccessException | InvocationTargetException | ClassNotFoundException e) {

--- a/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
@@ -4,6 +4,7 @@ import java.lang.reflect.Method;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Collections;
+import java.util.Map;
 import java.util.Set;
 import java.util.function.Consumer;
 
@@ -62,13 +63,14 @@ public class GenerateCodeMojo extends QuarkusBootstrapMojo {
             final Class<?> codeGenerator = deploymentClassLoader.loadClass("io.quarkus.deployment.CodeGenerator");
             final Method initAndRun = codeGenerator.getMethod("initAndRun", ClassLoader.class, Set.class, Path.class,
                     Path.class,
-                    Consumer.class, AppModel.class);
+                    Consumer.class, AppModel.class, Map.class);
             initAndRun.invoke(null, deploymentClassLoader,
                     Collections.singleton(sourcesDir),
                     generatedSourcesDir(test),
                     buildDir().toPath(),
                     sourceRegistrar,
-                    curatedApplication.getAppModel());
+                    curatedApplication.getAppModel(),
+                    mavenProject().getProperties());
         } catch (Exception any) {
             throw new MojoExecutionException("Quarkus code generation phase has failed", any);
         } finally {

--- a/docs/src/main/asciidoc/grpc-getting-started.adoc
+++ b/docs/src/main/asciidoc/grpc-getting-started.adoc
@@ -51,7 +51,8 @@ If you wish to generate code from different `proto` files for tests, also add th
 
 With this configuration, you can put your service and message definitions in the `src/main/proto` directory.
 `quarkus-maven-plugin` will generate Java files from your `proto` files.
-Alternatively to using the `prepare` goal of the `quarkus-maven-plugin`, you can use `protobuf-maven-plugin` to generate these files, more in <<Generating Java files from proto with protobuf-maven-plugin>>
+
+Alternatively to using the `generate-code` goal of the `quarkus-maven-plugin`, you can use `protobuf-maven-plugin` to generate these files, more in <<Generating Java files from proto with protobuf-maven-plugin>>
 
 Let's start with a simple _Hello_ service.
 Create the `src/main/proto/helloworld.proto` file with the following content:
@@ -95,7 +96,7 @@ $ mvn compile
 
 Once generated, you can look at the `target/generated-sources/grpc` directory:
 
-[source, txt]
+[source,txt]
 ----
 target/generated-sources/grpc
 └── io
@@ -114,6 +115,25 @@ These are the classes we are going to use.
 
 IMPORTANT: Every time you update the `proto` files, you need to re-generate the classes (using `mvn compile`).
 
+=== `proto` files with imports
+
+Protocol Buffers specification provides a way to import `proto` files.
+The Quarkus code generation mechanism lets you control the scope of dependencies to scan for possible imports by setting the `quarkus.generate-code.grpc.scan-for-imports` property to one of the following:
+
+- `all` - scan all the dependencies
+- `none` - don't scan the dependencies, use only what is defined in the `src/main/proto` or `src/test/proto`
+- `groupId1:artifactId1,groupId2:artifactId2` - scan only the dependencies with group id and artifact id in the list.
+
+If not specified, the property is set to `com.google.protobuf:protobuf-java`.
+To override it, set the `quarkus.generate-code.grpc.scan-for-imports` property in your pom.xml (or gradle.properties) to the desired value, e.g.
+
+[source,xml]
+----
+<properties>
+  <quarkus.generate-code.grpc.scan-for-imports>all</quarkus.generate-code.grpc.scan-for-imports>
+</properties>
+----
+
 == Implementing a gRPC service
 
 Now that we have the generated classes let's implement our _hello_ service.
@@ -126,7 +146,7 @@ IMPORTANT: Don't use `@ApplicationScoped` as the gRPC service implementation can
 
 Create the `src/main/java/org/acme/HelloService.java` file with the following content:
 
-[source, java]
+[source,java]
 ----
 package org.acme;
 

--- a/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java
+++ b/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java
@@ -2,6 +2,7 @@ package io.quarkus.grpc.deployment;
 
 import static java.lang.Boolean.TRUE;
 import static java.util.Arrays.asList;
+import static org.codehaus.plexus.util.FileUtils.copyStreamToFile;
 
 import java.io.BufferedWriter;
 import java.io.IOException;
@@ -9,9 +10,19 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
 import java.util.stream.Collectors;
 
+import org.codehaus.plexus.util.io.RawInputStreamFacade;
 import org.jboss.logging.Logger;
 
 import io.quarkus.bootstrap.model.AppArtifact;
@@ -22,13 +33,19 @@ import io.quarkus.bootstrap.prebuild.CodeGenFailureException;
 import io.quarkus.deployment.CodeGenContext;
 import io.quarkus.deployment.CodeGenProvider;
 import io.quarkus.deployment.util.ProcessUtil;
+import io.quarkus.runtime.util.HashUtil;
 import io.quarkus.utilities.JavaBinFinder;
 import io.quarkus.utilities.OS;
 
+/**
+ * Code generation for gRPC. Generates java classes from proto files placed in either src/main/proto or src/test/proto
+ * Inspired by <a href="https://github.com/xolstice/protobuf-maven-plugin">Protobuf Maven Plugin</a>
+ */
 public class GrpcCodeGen implements CodeGenProvider {
     private static final Logger log = Logger.getLogger(GrpcCodeGen.class);
 
     private static final String quarkusProtocPluginMain = "io.quarkus.grpc.protoc.plugin.MutinyGrpcGenerator";
+    private static final String PROTO = ".proto";
 
     private Executables executables;
 
@@ -61,16 +78,22 @@ public class GrpcCodeGen implements CodeGenProvider {
             if (Files.isDirectory(protoDir)) {
                 List<String> protoFiles = Files.walk(protoDir)
                         .filter(Files::isRegularFile)
-                        .filter(path -> path.toString().endsWith(".proto"))
+                        .filter(path -> path.toString().endsWith(PROTO))
                         .map(Path::toString)
                         .map(this::escapeWhitespace)
                         .collect(Collectors.toList());
                 if (!protoFiles.isEmpty()) {
                     initExecutables(workDir, context.appModel());
 
+                    Collection<String> protosToImport = gatherImports(workDir.resolve("protoc-dependencies"), context);
+
                     List<String> command = new ArrayList<>();
-                    command.addAll(asList(executables.protoc.toString(),
-                            "-I=" + escapeWhitespace(protoDir.toString()),
+                    command.add(executables.protoc.toString());
+                    for (String protoImportDir : protosToImport) {
+                        command.add(String.format("-I=%s", escapeWhitespace(protoImportDir)));
+                    }
+
+                    command.addAll(asList("-I=" + escapeWhitespace(protoDir.toString()),
                             "--plugin=protoc-gen-grpc=" + executables.grpc,
                             "--plugin=protoc-gen-q-grpc=" + executables.quarkusGrpc,
                             "--q-grpc_out=" + outDir,
@@ -93,6 +116,68 @@ public class GrpcCodeGen implements CodeGenProvider {
             throw new CodeGenException("Failed to generate java files from proto file in " + protoDir.toAbsolutePath(), e);
         }
         return false;
+    }
+
+    private Collection<String> gatherImports(Path workDir, CodeGenContext context) throws CodeGenException {
+        Map<String, String> properties = context.properties();
+
+        String scanForImports = properties.getOrDefault("quarkus.generate-code.grpc.scan-for-imports",
+                "com.google.protobuf:protobuf-java");
+
+        if ("none".equals(scanForImports.toLowerCase(Locale.getDefault()))) {
+            return Collections.emptyList();
+        }
+
+        boolean scanAll = "all".equals(scanForImports.toLowerCase(Locale.getDefault()));
+        List<String> dependenciesToScan = Arrays.asList(scanForImports.split(","));
+
+        Set<String> importDirectories = new HashSet<>();
+        AppModel appModel = context.appModel();
+        for (AppDependency dependency : appModel.getUserDependencies()) {
+            AppArtifact artifact = dependency.getArtifact();
+            if (scanAll
+                    || dependenciesToScan.contains(String.format("%s:%s", artifact.getGroupId(), artifact.getArtifactId()))) {
+                for (Path path : artifact.getPaths()) {
+                    Path jarName = path.getFileName();
+                    if (jarName.toString().endsWith(".jar")) {
+                        final JarFile jar;
+                        try {
+                            jar = new JarFile(path.toFile());
+                        } catch (final IOException e) {
+                            throw new CodeGenException("Failed to read Jar: " + path.normalize().toAbsolutePath(), e);
+                        }
+
+                        for (JarEntry jarEntry : jar.stream().collect(Collectors.toList())) {
+                            String jarEntryName = jarEntry.getName();
+                            if (jarEntryName.endsWith(PROTO)) {
+                                Path protoUnzipDir = workDir.resolve(HashUtil.sha1(jar.getName())).normalize().toAbsolutePath();
+                                try {
+                                    Files.createDirectories(protoUnzipDir);
+                                    importDirectories.add(protoUnzipDir.toString());
+                                } catch (IOException e) {
+                                    throw new CodeGenException(
+                                            "Failed to create directory: " + protoUnzipDir, e);
+                                }
+                                // checking for https://snyk.io/research/zip-slip-vulnerability
+                                Path unzippedProto = protoUnzipDir.resolve(jarEntryName);
+                                if (!unzippedProto.normalize().toAbsolutePath().startsWith(workDir.toAbsolutePath())) {
+                                    throw new CodeGenException("Attempted to unzip " + jarEntryName
+                                            + " to a location outside the working directory");
+                                }
+                                try {
+                                    copyStreamToFile(new RawInputStreamFacade(jar.getInputStream(jarEntry)),
+                                            unzippedProto.toFile());
+                                } catch (IOException e) {
+                                    throw new CodeGenException("Failed to create input stream for reading " + jarEntryName
+                                            + " from " + jar.getName(), e);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return importDirectories;
     }
 
     private String escapeWhitespace(String path) {

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/GrpcWithIncludesDevModeTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/GrpcWithIncludesDevModeTest.java
@@ -1,0 +1,31 @@
+package io.quarkus.gradle.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Disabled;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * A test to check if a proto file that imports another proto from dependencies
+ * works properly in dev mode if quarkus.generate-code.grpc.scan-for-imports=true is specified
+ */
+@Disabled
+public class GrpcWithIncludesDevModeTest extends QuarkusDevGradleTestBase {
+    @Override
+    protected String projectDirectoryName() {
+        return "grpc-include-project";
+    }
+
+    @Override
+    protected void testDevMode() throws Exception {
+        assertThat(getHttpResponse("/hello")).isEqualTo("hello 2");
+
+        replace("application/src/main/proto/hello.proto",
+                ImmutableMap.of("TEST_ONE = 2;", "TEST_ONE = 15;"));
+
+        Thread.sleep(1000);
+
+        assertThat(getHttpResponse("/hello")).isEqualTo("hello 15");
+    }
+}

--- a/integration-tests/gradle/src/test/resources/grpc-include-project/build.gradle
+++ b/integration-tests/gradle/src/test/resources/grpc-include-project/build.gradle
@@ -1,0 +1,31 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+repositories {
+    jcenter()
+    mavenLocal()
+    mavenCentral()
+    gradlePluginPortal()
+}
+
+dependencies {
+    implementation 'io.quarkus:quarkus-grpc'
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+}
+
+group 'org.acme'
+version '1.0.0-SNAPSHOT'
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+test {
+    systemProperty "java.util.logging.manager", "org.jboss.logmanager.LogManager"
+}

--- a/integration-tests/gradle/src/test/resources/grpc-include-project/gradle.properties
+++ b/integration-tests/gradle/src/test/resources/grpc-include-project/gradle.properties
@@ -1,0 +1,3 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus
+quarkus.generate-code.grpc.scan-for-imports=true

--- a/integration-tests/gradle/src/test/resources/grpc-include-project/settings.gradle
+++ b/integration-tests/gradle/src/test/resources/grpc-include-project/settings.gradle
@@ -1,0 +1,11 @@
+pluginManagement {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    plugins {
+        id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+rootProject.name = 'grpc-include-project'

--- a/integration-tests/gradle/src/test/resources/grpc-include-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
+++ b/integration-tests/gradle/src/test/resources/grpc-include-project/src/main/java/org/acme/quarkus/sample/HelloResource.java
@@ -1,0 +1,21 @@
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.example.HelloMsg;
+
+@Path("/hello")
+public class HelloResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        Integer number = HelloMsg.Status.TEST_ONE.getNumber();
+        // return a thing from proto file (for devmode test)
+        return "hello " + number;
+    }
+}

--- a/integration-tests/gradle/src/test/resources/grpc-include-project/src/main/proto/hello.proto
+++ b/integration-tests/gradle/src/test/resources/grpc-include-project/src/main/proto/hello.proto
@@ -1,0 +1,23 @@
+syntax = "proto3";
+package io.quarkus.example;
+
+option java_multiple_files = true;
+option java_package = "io.quarkus.example";
+
+import "google/protobuf/timestamp.proto";
+
+
+message HelloMsg {
+  enum Status {
+    UNKNOWN = 0;
+    NOT_SERVING = 1;
+    TEST_ONE = 2;
+  }
+  string message = 1;
+  google.protobuf.Timestamp date_time = 2;
+  Status status = 3;
+}
+
+service DevModeService {
+  rpc Check(HelloMsg) returns (HelloMsg);
+}

--- a/integration-tests/grpc-plain-text/src/main/proto/helloworld.proto
+++ b/integration-tests/grpc-plain-text/src/main/proto/helloworld.proto
@@ -36,6 +36,8 @@ option objc_class_prefix = "HLW";
 
 package helloworld;
 
+import "google/protobuf/timestamp.proto";
+
 // The greeting service definition.
 service Greeter {
     // Sends a greeting
@@ -45,6 +47,7 @@ service Greeter {
 // The request message containing the user's name.
 message HelloRequest {
     string name = 1;
+    google.protobuf.Timestamp date_time = 2;
 }
 
 // The response message containing the greetings

